pkg/utils/workflow.go

```diff
@@ -120,41 +121,68 @@ func sliceContains(s []string, e string) bool {
-				chaosEngines = append(chaosEngines, a.Name)
-			}
-		}
 
-		// Template that contains ChaosEngine manifest
-		if sliceContains(chaosEngines, t.Name) {
-			var weightageInput model.WeightagesInput
-			var err error
+		var err error
 
-			var chaosEngine chaosTypes.ChaosEngine
-			err = yaml.Unmarshal([]byte(t.Inputs.Artifacts[0].Raw.Data), &chaosEngine)
-			if err != nil {
-				return errors.New("Error parsing ChaosEngine: " + err.Error())
+		if t.Inputs.Artifacts != nil && len(t.Inputs.Artifacts) > 0 {
+			if t.Inputs.Artifacts[0].Raw == nil {
+				continue
 			}
-			weightageInput.ExperimentName = chaosEngine.ObjectMeta.GenerateName
-			w, ok := t.Metadata.Labels["weight"]
 
-			if !ok {
-				White.Println("Weightage for ChaosExperiment/" + weightageInput.ExperimentName + " not provided, defaulting to 10.")
-				w = "10"
-			}
-			weightageInput.Weightage, err = strconv.Atoi(w)
-			if err != nil {
-				return errors.New("Invalid weightage for ChaosExperiment/" + weightageInput.ExperimentName + ".")
-			}
+			var data = t.Inputs.Artifacts[0].Raw.Data
+			if len(data) > 0 {
+				// This replacement is required because chaos engine yaml have a syntax template. example:{{ workflow.parameters.adminModeNamespace }}
+				// And it is not able the unmarshal the yamlstring to chaos engine struct
+				data = strings.ReplaceAll(data, "{{", "")
+				data = strings.ReplaceAll(data, "}}", "")
+
+				var chaosEngine chaosTypes.ChaosEngine
+
+				err = yaml.Unmarshal([]byte(data), &chaosEngine)
+
+				if err != nil {
+					return errors.New("failed to unmarshal chaosengine")
+				}
+
+				if strings.ToLower(chaosEngine.Kind) == "chaosengine" {
+					var weightageInput model.WeightagesInput
 
-			chaosWorkFlowRequest.Weightages = append(chaosWorkFlowRequest.Weightages, &weightageInput)
+					weightageInput.ExperimentName = chaosEngine.ObjectMeta.GenerateName
+
+					if len(weightageInput.ExperimentName) == 0 {
+						return errors.New("empty chaos experiment name")
+					}
+
+					if len(chaosEngine.Spec.Experiments) == 0 {
+						return errors.New("no experiments specified in chaosengine - " + weightageInput.ExperimentName)
+					}
+
+					w, ok := t.Metadata.Labels["weight"]
+
+					if !ok {
+						White.Println("Weightage for ChaosExperiment/" + weightageInput.ExperimentName + " not provided, defaulting to 10.")
+						w = "10"
+					}
+					weightageInput.Weightage, err = strconv.Atoi(w)
+
+					if err != nil {
+						return errors.New("Invalid weightage for ChaosExperiment/" + weightageInput.ExperimentName + ".")
+					}
+
+					chaosWorkFlowRequest.Weightages = append(chaosWorkFlowRequest.Weightages, &weightageInput)
+				}
+			}
 		}
 	}
+
+	// If no experiments are present in the workflow, adds a 0 to the Weightages array so it doesn't fail (same behaviour as the UI)
+	if len(chaosWorkFlowRequest.Weightages) == 0 {
+		White.Println("No experiments found in the workflow, defaulting experiments weightage to 0.")
```

Comment:
Could you add this minor terminology change?
```suggestion
		White.Println("No experiments found in the scenario, defaulting experiments weightage to 0.")
```